src/components/ProjectSettingsTabs/NotificationsTab/getFormSchema.ts

```diff
@@ -0,0 +1,72 @@
+import { Field } from "@rjsf/core";
+import { SpruceFormProps } from "components/SpruceForm";
+import { CardFieldTemplate } from "components/SpruceForm/FieldTemplates";
+import widgets from "components/SpruceForm/Widgets";
+import { radioBoxOptions } from "../utils";
+import { DefaultSubscriptionsField } from "./Fields/DefaultSubscriptionsField";
+
+export const getFormSchema = (
+  useRepoSettings: boolean,
+  repoData?: any
+): {
+  fields: Record<string, Field>;
+  schema: SpruceFormProps["schema"];
+  uiSchema: SpruceFormProps["uiSchema"];
+} => ({
+  fields: {
+    defaultSubscriptionsField: DefaultSubscriptionsField,
+  },
+  schema: {
+    type: "object" as "object",
+    properties: {
+      buildBreakSettings: {
+        type: "object" as "object",
+        title: "Notifications",
+        properties: {
+          notifyOnBuildFailure: {
+            type: ["boolean", "null"],
+            title: "Build-break Notifications",
+            oneOf: radioBoxOptions(
+              ["Enabled", "Disabled"],
+              repoData?.buildBreakSettings?.notifyOnBuildFailure
+            ),
+          },
+        },
+      },
+      defaultSubscriptions: {
+        type: "object" as "object",
+        title: "Subscriptions",
+        properties: {
+          defaultToRepo: {
+            type: ["array", "null"],
+            items: {
+              type: "object" as "object",
+              properties: {
+                filePattern: {
+                  type: "string" as "string",
+                  title: "TODO: EVG-16262 - this is not implemented yet",
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+  uiSchema: {
+    buildBreakSettings: {
+      "ui:rootFieldId": "notifications",
+      "ui:ObjectFieldTemplate": CardFieldTemplate,
+      notifyOnBuildFailure: {
+        "ui:widget": widgets.RadioBoxWidget,
+      },
+    },
+    defaultSubscriptions: {
+      defaultToRepo: {
+        "ui:addButtonText": "Add New Subscription",
+        "ui:field": "defaultSubscriptionsField",
```

Comment:
Rather than creating custom fields for this pattern, I added [this `overrideRadioBox` helper](https://github.com/evergreen-ci/spruce/blob/ca679972e64b3d043b2d358a5fc8a7a687f05ad2/src/components/ProjectSettingsTabs/utils.ts#L44) that leverages more of the data validation features provided by our form package. There's an example in use [here](https://github.com/evergreen-ci/spruce/blob/ca679972e64b3d043b2d358a5fc8a7a687f05ad2/src/components/ProjectSettingsTabs/GithubCommitQueueTab/getFormSchema.ts#L149-L160), and the data typing should look something like [this](https://github.com/evergreen-ci/spruce/blob/ca679972e64b3d043b2d358a5fc8a7a687f05ad2/src/components/ProjectSettingsTabs/GithubCommitQueueTab/types.ts#L33-L39) (the `Array<string>` can be replaced with the actual shape of your data of course).

Reply:
thanks!